application/src/main/java/org/opentripplanner/transit/api/request/TripTimeOnDateRequest.java

```diff
@@ -0,0 +1,110 @@
+package org.opentripplanner.transit.api.request;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Objects;
+import java.util.Set;
+import org.opentripplanner.model.TripTimeOnDate;
+import org.opentripplanner.transit.api.model.FilterValues;
+import org.opentripplanner.transit.model.basic.TransitMode;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.model.site.StopLocation;
+import org.opentripplanner.transit.service.ArrivalDeparture;
+
+public class TripTimeOnDateRequest {
+
+  private final Set<StopLocation> stopLocations;
```

Comment:
A request should not contain domain entities  - these would need to be looked up before calling the domain service.

Reply:
We talked about it in the dev meeting and the reason is that many methods in TransitService already use the StopLocation as the input type. Many underlying indices also use it as the key.